node/flags/deprecated_flag.go

```diff
@@ -0,0 +1,23 @@
+package flags
+
+type deprecatedFlag struct {
+	// error to return when flag is used
+	err error
```

Comment:
This can just be a string

Reply:
>That would at least remove the need to disable linters everywhere we use deprecatedFlag 🙂

It would be "cheating" the linter... (changing the code so it cannot detect the issue).

>have a Hint() / Description() or similar method that is then called with panic in EnsureCLIFlags.

I'd rather not panic on invalid user arguments. It's bad UX. The app should print the help message + clear information about what was invalid and optionally how to fix it at the bottom so it's easy to notice.

Reply:
> It would be "cheating" the linter... (changing the code so it cannot detect the issue).
I rather see it the other way round, we are abusing `error` as a log message that's why the linter is complaining in the first place.

Reply:
That being said, I am OK with the current approach - the global logger is a bit annoying but changing the code to be able to pass an instance of the logger is out of scope for this PR; we should do that at a future point though.